src/Compilers/Core/Rebuild/CompilationOptionsReader.cs

```diff
@@ -340,16 +340,34 @@ private static IEnumerable<MetadataReferenceInfo> ParseMetadataReferenceInfo(Blo
                 var imageSize = blobReader.ReadInt32();
                 var mvid = blobReader.ReadGuid();
 
-                yield return new MetadataReferenceInfo(
-                    timestamp,
-                    imageSize,
-                    name,
-                    mvid,
-                    string.IsNullOrEmpty(externAliases)
-                        ? ImmutableArray<string>.Empty
-                        : externAliases.Split(',').ToImmutableArray(),
-                    kind,
-                    embedInteropTypes);
+                if (string.IsNullOrEmpty(externAliases))
+                {
+                    yield return new MetadataReferenceInfo(
+                        timestamp,
+                        imageSize,
+                        name,
+                        mvid,
+                        externAlias: null,
+                        kind,
+                        embedInteropTypes);
+                }
+                else
+                {
+                    foreach (var alias in externAliases.Split(','))
+                    {
+                        // The "global" alias is an invention of the tooling on top of the compiler. 
+                        // The compiler itself just sees "global" as a reference without any aliases
+                        // and we need to mimic that here.
+                        yield return new MetadataReferenceInfo(
+                            timestamp,
+                            imageSize,
+                            name,
+                            mvid,
+                            alias == "global" ? null : alias,
```

Comment:
does this constant live anywhere in the compiler that we can reference from here?

Reply:
Not that I could find and I couldn't find a place that made sense as a place to define so that it would be usable across all the layers of the compiler.